src/monda/sorad/qc.py

```diff
@@ -43,7 +50,36 @@ def nearest(items, pivot):
      return nearest_value, nearest_index
 
 
-# Filters used in QC of l and e spectra (Step 1 in 3C and FP chain)
+# Filters used in Step (0) QC (based on tilt and relative azimuth)
+def rel_az_filter(rel_view_az, lower_azi_bound = 110, upper_azi_bound = 150):
+    """ Function to test that So-Rad relative viewing azimiuth is within given angular bounds.  """    
+
+    q_lower =  ~ np.isnan(np.where(np.abs(rel_view_az) < upper_azi_bound, rel_view_az, np.nan))
+    q_upper =  ~ np.isnan(np.where(np.abs(rel_view_az) > lower_azi_bound, rel_view_az, np.nan))
+    q_az =  np.logical_and(q_lower, q_upper)
+    q_az =  np.ravel(q_az.astype(int))
+
+    return q_az
+
+def tilt_filter(tilt_avgs, tilt_stds, upper_tilt_bound=5, upper_tilt_std_bound = 2):
+    """ Applies upper thresholds to allowed tilt and tilt-standard-deviation values (degs) """
+
+    mode_tilt_avgs = mode(tilt_avgs)   # test for constant values  
+    number_constant = len(tilt_avgs[mode_tilt_avgs == tilt_avgs])
+       
+    if (number_constant > 1) & (len(tilt_avgs) > 1):
+        log.error("Tilt sensor was not applied  (stuck on single value)")
```

Comment:
should this behaviour be user-defined rather than default to a pass

Reply:
yeh - good point. I defaulted to  pass for the Tara Ocean, (as I also normally had co-incident wind data). I have therefore default to a NaN  in the QC field if the sensor is stuck